openerp/tools/cache.py

```diff
@@ -89,52 +116,54 @@ def clear(self, model, *args):
 
 class ormcache_context(ormcache):
     def __init__(self, skiparg=2, size=8192, accepted_keys=()):
-        super(ormcache_context,self).__init__(skiparg,size)
-        self.accepted_keys = accepted_keys
-
-    def __call__(self, method):
-        # remember which argument is context
-        args = getargspec(method)[0]
-        self.context_pos = args.index('context')
-        return super(ormcache_context, self).__call__(method)
-
-    def lookup(self, method, *args, **kwargs):
-        d, key0 = self.lru(args[0])
-
-        # Note. The decorator() wrapper (used in __call__ above) will resolve
-        # arguments, and pass them positionally to lookup(). This is why context
-        # is not passed through kwargs!
-        if self.context_pos < len(args):
-            context = args[self.context_pos]
-        else:
-            context = kwargs.get('context') or {}
-        ckey = [(k, context[k]) for k in self.accepted_keys if k in context]
-
-        # Beware: do not take the context from args!
-        key = key0 + args[self.skiparg:self.context_pos] + tuple(ckey)
-        try:
-            r = d[key]
-            STAT[key0].hit += 1
-            return r
-        except KeyError:
-            STAT[key0].miss += 1
-            value = d[key] = self.method(*args, **kwargs)
-            return value
-        except TypeError:
-            STAT[key0].err += 1
-            return self.method(*args, **kwargs)
+        raise NotImplementedError("ormcache_context() is no longer available; it has been superseded by ormcache()")
 
 
 class ormcache_multi(ormcache):
-    def __init__(self, skiparg=2, size=8192, multi=3):
-        assert skiparg <= multi
-        super(ormcache_multi, self).__init__(skiparg, size)
-        self.multi = multi
+    def __init__(self, *args, **kwargs):
+        super(ormcache_multi, self).__init__(*args, **kwargs)
+        self.multi = kwargs.get('multi')
+        assert self.multi is not None, "Expected keyword argument multi=..."
```

Comment:
Couldn't `ormcache(multi=)` automatically delegate to ormcache_multi's implementation? That way no need to bother with two interfaces.